src/pke/include/cryptocontext.h

```diff
@@ -2372,11 +2387,19 @@ class CryptoContextImpl : public Serializable {
     void ModReduceInPlace(Ciphertext<Element>& ciphertext) const {
         ValidateCiphertext(ciphertext);
 
-        GetScheme()->ModReduceInPlace(ciphertext, BASE_NUM_LEVELS_TO_DROP);
+        if (isCKKS(m_schemeId)) {
```

Comment:
Any CKKS-specific code should be implemented inside the folder for the CKKS scheme (not in global cryptocontext code)

Reply:
Fixed.